libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -422,213 +423,261 @@ export class DynamicFormGroup<TModel> extends FormGroup {
+      if (isPropertyValidatorOfType(validationMetadata, fieldName, ValidationKeys.nested.type)) {
+        nestedGroupValidations.push(validationMetadata);
+      }
+    });
+
+    const fieldDefinition: DynamicFormGroupField = {
+      data: formGroupFields[fieldName],
+      validationFunctions: [],
+      validationDefinitions: []
+    };
+
+    if (fieldDefinition.data === undefined) {
+      fieldDefinition.data = fields[fieldName];
+    }
+
+    validationGroupMetadatas.forEach(validationMetadata => {
+      if (validationMetadata.propertyName === fieldName && validationMetadata.type !== ValidationKeys.conditional.type) {
+
+        // Add all validation to the field except the @NestedValidation definition as
+        // being part of the form would imply it is validated if any other rules are present
+        if (validationMetadata.type !== ValidationKeys.nested.type) {
+          fieldDefinition.validationDefinitions.push(validationMetadata);
+        }
+
+        for (const typeKey in ValidationTypes) {
+          if (ValidationTypes.hasOwnProperty(typeKey)) {
+
+            // Handle Nested Validation
+            if (checkWithAllNestedValidations(allNestedValidations, nestedGroupValidations, fieldName)) {
+              if (isNestedValidate(validationMetadata, typeKey)) {
+
+                const objectToValidate = fields[fieldName] instanceof DynamicFormGroup ? fields[fieldName].object : undefined;
+                const nestedValidate = createNestedValidate(objectToValidate, validationMetadata);
+
+                setFieldData(fieldName, fieldDefinition, nestedValidate);
               }
             }
+
+            // Handle Custom Validation
+            if (isCustomValidate(validationMetadata, typeKey)) {
+              const customValidation = createCustomValidation(fieldName, validationMetadata);
+              setFieldData(fieldName, fieldDefinition, customValidation);
+            }
+
+            // Handle remaining validation
+            if (isDynamicValidate(validationMetadata, typeKey)) {
+              const dynamicValidate = createDynamicValidate(validationMetadata, conditionalValidations, fieldName);
+              setFieldData(fieldName, fieldDefinition, dynamicValidate);
+            }
           }
         }
-      });
-      if (formGroupField[0] instanceof DynamicFormGroup) {
-        formGroupFields[key] = formGroupField[0];
-      } else {
-        if (formGroupField[0] instanceof FormArray) {
-          formGroupFields[key] = formGroupField[0];
-        } else {
-          formGroupFields[key] = [formGroupField[0], formGroupField.filter((item, index) => index !== 0)];
-        }
       }
     });
+
+    // Convert to a structure, angular understands
+    if (fieldDefinition.data instanceof DynamicFormGroup || fieldDefinition.data instanceof FormArray) {
+      formGroupFields[fieldName] = fieldDefinition.data;
+    }
+    else {
+      formGroupFields[fieldName] = new DynamicFormControl(fieldDefinition);
```

Comment:
Core feature code. Returning the control directly works for all instance that it would be added to a group or array